src/OpenTelemetry.AutoInstrumentation.NLogTarget/OpenTelemetryTarget.cs

```diff
@@ -0,0 +1,211 @@
+        var properties = new List<KeyValuePair<string, object?>>();
+        if (IncludeEventProperties && logEvent.HasProperties && logEvent.Properties is not null)
+        {
+            foreach (var kvp in logEvent.Properties)
+            {
+                properties.Add(new KeyValuePair<string, object?>(Convert.ToString(kvp.Key)!, kvp.Value));
+            }
+        }
+
+        // Scope properties can be added via explicit <attribute> entries or NLog's contexts (GDC/MDLC)
+        foreach (var attribute in Attributes)
+        {
+            var value = attribute.Layout?.Render(logEvent);
+            if (!string.IsNullOrEmpty(attribute.Name))
+            {
+                properties.Add(new KeyValuePair<string, object?>(attribute.Name!, value));
+            }
+        }
+
+        var body = IncludeFormattedMessage ? logEvent.FormattedMessage : Convert.ToString(logEvent.Message);
+
+        var severityText = logEvent.Level.Name;
+        var severityNumber = MapLogLevelToSeverity(logEvent.Level);
+
+        var current = Activity.Current;
+
+        // Emit using internal helpers via reflection delegate
+        var renderedMessage = logEvent.FormattedMessage;
+        var args = IncludeEventParameters && logEvent.Parameters is object[] p ? p : null;
+
+        OpenTelemetry.AutoInstrumentation.Instrumentations.NLog.Bridge.OpenTelemetryLogHelpers.LogEmitter?.Invoke(
+            logger,
+            body,
+            logEvent.TimeStamp,
+            severityText,
+            severityNumber,
+            logEvent.Exception,
+            properties,
+            current,
+            args,
+            renderedMessage);
+    }
+
+    private static int MapLogLevelToSeverity(LogLevel level)
+    {
+        // Map NLog ordinals 0..5 to OTEL severity 1..24 approximate buckets
+        return level.Ordinal switch
+        {
+            0 => 1,   // Trace
+            1 => 5,   // Debug
+            2 => 9,   // Info
+            3 => 13,  // Warn
+            4 => 17,  // Error
+            5 => 21,  // Fatal
+            _ => 9
+        };
+    }
+
+    private static Func<string?, object?>? CreateGetLoggerDelegate(LoggerProvider loggerProvider)
+    {
+        try
+        {
+            var methodInfo = typeof(LoggerProvider)
+                .GetMethod("GetLogger", System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance, null, new[] { typeof(string) }, null)!;
+            return (Func<string?, object?>)methodInfo.CreateDelegate(typeof(Func<string?, object?>), loggerProvider);
+        }
+        catch
+        {
+            return null;
+        }
+    }
+
+    private object GetOrCreateLogger(string? loggerName)
+    {
+        var key = loggerName ?? string.Empty;
+        if (LoggerCache.TryGetValue(key, out var logger))
+        {
+            return logger;
+        }
+
+        var factory = _getLoggerFactory;
+        if (factory is null)
+        {
+            return new object();
```

Comment:
This looks weird. Why not allow `object?` and just discards LogEvents when `GetOrCreateLogger` returns `null` ?